### ISSUE TITLE
Playback progress not reported to the PMS

### ISSUE THREAD

**Issue by croneter** [croneter/PlexKodiConnect#1772]
```
2022-01-09 17:00:54.040 T:5448     INFO <general>: PLEX.companion.playstate: ----===## PlaystateMgr stopped ##===----
2022-01-09 17:00:54.040 T:5448    ERROR <general>: Exception in thread 
2022-01-09 17:00:54.040 T:5448    ERROR <general>: Thread-5
2022-01-09 17:00:54.040 T:5448    ERROR <general>: :
                                                   
2022-01-09 17:00:54.040 T:5448    ERROR <general>: Traceback (most recent call last):
                                                   
2022-01-09 17:00:54.040 T:5448    ERROR <general>:   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
                                                   
2022-01-09 17:00:54.041 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/plex_companion/playstate.py", line 195, in run
                                                   
2022-01-09 17:00:54.044 T:5448    ERROR <general>:     
2022-01-09 17:00:54.044 T:5448    ERROR <general>: self._run()
2022-01-09 17:00:54.044 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.044 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/plex_companion/playstate.py", line 248, in _run
                                                   
2022-01-09 17:00:54.045 T:5448    ERROR <general>:     
2022-01-09 17:00:54.045 T:5448    ERROR <general>: message = timeline(players)
2022-01-09 17:00:54.045 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.045 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/plex_companion/common.py", line 203, in timeline
                                                   
2022-01-09 17:00:54.047 T:5448    ERROR <general>:     
2022-01-09 17:00:54.047 T:5448    ERROR <general>: timeline = timeline_dict(player['playerid'], typus)
2022-01-09 17:00:54.047 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.047 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/plex_companion/common.py", line 169, in timeline_dict
                                                   
2022-01-09 17:00:54.047 T:5448    ERROR <general>:     
2022-01-09 17:00:54.048 T:5448    ERROR <general>: item.current_kodi_video_stream = info['currentvideostream']['index']
2022-01-09 17:00:54.048 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.048 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/playlist_func.py", line 122, in current_kodi_video_stream
                                                   
2022-01-09 17:00:54.048 T:6280    DEBUG <general>: CVideoPlayerVideo::OutputPicture - ttd:1370ms pts:1.070 Clock:-0.301 Level:-1
2022-01-09 17:00:54.048 T:6280  WARNING <general>: OutputPicture - timeout waiting for buffer
2022-01-09 17:00:54.048 T:6280    DEBUG <general>: CVideoPlayerVideo - CDVDMsg::DEMUXER_PACKET - checking interlace vfmt: interlace
2022-01-09 17:00:54.050 T:6280    DEBUG <general>: CRenderManager::Configure - change configuration. 720x576. display: 720x404. framerate: 50.00.
2022-01-09 17:00:54.050 T:5448    ERROR <general>:     
2022-01-09 17:00:54.050 T:5448    ERROR <general>: self.on_kodi_video_stream_change(value)
2022-01-09 17:00:54.050 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.050 T:5448    ERROR <general>:   File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/playlist_func.py", line 354, in on_kodi_video_stream_change
                                                   
2022-01-09 17:00:54.051 T:5448    ERROR <general>:     
2022-01-09 17:00:54.051 T:5448    ERROR <general>: plex_stream_index = int(self.video_streams[kodi_stream_index].get('id'))
2022-01-09 17:00:54.051 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.051 T:5448    ERROR <general>: IndexError
2022-01-09 17:00:54.051 T:5448    ERROR <general>: : 
2022-01-09 17:00:54.051 T:5448    ERROR <general>: list index out of range
2022-01-09 17:00:54.051 T:5448    ERROR <general>: 
                                                   
2022-01-09 17:00:54.051 T:5448    DEBUG <general>: .
2022-01-09 17:00:54.069 T:5292     INFO <general>: Skipped 1 duplicate messages..
```

**Comment by pbo10** [comment]
Ah there's a known issue with the progress not saving to PMS. I've been having this issue on and off but happened to migrate my PMS to a new host at around the same time it started so I thought it was related to that. Glad I'm not the only one, for me it's not every time, sometimes it saves progress and others it doesn't.

**Comment by croneter** [comment]
@pbo10 
DEBUG log files of this issue occuring would be extremly helpful!

**Comment by pbo10** [comment]
@croneter Sure I'll try and grab some tonight, just the usual enable debugging in Kodi and save the log after I assume? 

I think manually selecting something and marking as watched/unwatched works fine, it only appears to be when you watch something either part way through or to the end and it doesn't always get marked as watched from what I've experienced so far.

**Comment by croneter** [comment]
Correct, just enable debug logging. And mind to delete the tokens before uploading anything, see https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by pbo10** [comment]
@croneter 
I wanted to spend some extra time testing this to make sure it's not an issue my end as I have 3x TVs with almost an identical set up on each, 2x running on Nvidia Shield Tubes using SD card as storage and 1x on an Nvidia Shield Pro using internal storage and I found the issue only seems to be present on the Nvidia Shield Pro though it is possible that's just because it gets more use than the others. I also wanted to make sure this wasn't just an issue on that one device that could be fixed easily with a simple reset etc so I did a reset of the database and confirmed the issue is still present after this.

From what I've seen the issue only happens when you actually watch something and expect it to be automatically marked as watched or if you stop half way through and expect the progress to be saved, what happens is Kodi shows what you would expect IE it gets marked as watched but this is not updated on the PMS so if you wait or exit the TV show page and go back in it changes after a while back to the status on the PMS (so it's reading from PMS but not writing to PMS). Interestingly though I've never seen it fail when you right click and mark as watched, this always updates the PMS every single time.

Grabbing the log with debugging enabled is proving to be more difficult, because restarting Kodi fixes the issue at least for a while and I don't know how to cause it to fail again so when I went to grab the log tonight I enabled debugging and restarted and it was working fine. I have a very long log file that I took before doing anything which has the same error message you've added above, and at this point it's not updating PMS at all but debugging wasn't enabled at the time and it's a very log long file as Kodi had been running for some time. Let me know if this log file would be useful and if there's an easy way to get this to you, I'm not sure I want to post the entire thing publicly.

I guess the only way I could get a full debug log is to leave debugging on 24/7 until it breaks again and it'll just be a massive log file as I can't restart Kodi.

I don't know enough about the plugin to try and understand this, but is PlaystateMgr a process that may have failed earlier in the day and restarting Kodi restarts this process so it works until the process fails again? And does right clicking and selecting mark as watch use a different process maybe, so that's why that always keeps working?

Also out of interest where did the log you originally posted come from? Is someone else having a similar issue?

**Comment by Speederer** [comment]
Hi, I have a similar (maybe the same) problem with PKC on a CoreElec device. I started to use it ~1 month ago and it was working fine at the beginning, but it hasn't updated the PMS database for 1 or 2 weeks. I'm not sure what caused/triggered this issue, maybe a CoreElec update. 
I've just checked this function in a Windows based Kodi and it looks fine.
"Mark as watched" button works fine in both case (Windows and CoreElec).

**Comment by croneter** [comment]
Thanks for the inputs, but I just need DEBUG log files to fix this. See https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by Speederer** [comment]
Hi,
I started the logging and after that I watched a series and a movie for ~2 minutes.
Furthermore I recognized that playing movies via PKC is not shown in the PMS Activity section. Normally I can follow the connected users, I mean what they are watching, but this function is not working from my CoreElec device which doesn't report back the status to the PMS.

[pkc_kodi.zip](https://github.com/croneter/PlexKodiConnect/files/7900498/pkc_kodi.zip)
.

**Comment by bjorn1981** [comment]
I've been experiencing this issue as wel. Here is my debug log, started a movie from a resume point and let run for e few minutes. Resume point is saved in Kodi but not in PMS.
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/7904480/kodi.log)

**Comment by gregeeh** [comment]
Would really like this issue fixed as everyone has to manually mark TV Shows as watched.

**Comment by croneter** [comment]
Does PKC beta version 3.6.4 help?

**Comment by gregeeh** [comment]
> Does PKC beta version 3.6.4 help?

Not using beta.  How can I swap to beta and how can I then go back to stable?

**Comment by croneter** [comment]
Just install the beta repository additionally and grab/install the new version manually. Then switch back by doing the same using the stable repo

**Comment by gregeeh** [comment]
Just viewed one TV Show and it was marked as watched in PMS and the View History was also updated correctly.  Seems fixed but will try more in the next couple of days.  Thank you.

**Comment by gregeeh** [comment]
Still on 3.6.4 beta and played one episode of 1883 last night.  See how Plex & Tautulli have reported it.  And no I did not play it twice.

[https://i.imgur.com/XAefizT.jpeg](url)

[https://i.imgur.com/31RSHmz.jpeg](url)

**Comment by croneter** [comment]
@gregeeh I can't see anything on your screenshots. You can just drop images here on GitHub into the reply form, much easier.

What exactly is the problem? That playback is reported twice instead of only once?

**Comment by gregeeh** [comment]
`What exactly is the problem? That playback is reported twice instead of only once?`

No 1.5 times.

![XAefizT](https://user-images.githubusercontent.com/46234729/151916998-cace5cd2-dca8-4019-aac7-d245620d3198.jpeg)
![31RSHmz](https://user-images.githubusercontent.com/46234729/151917006-81769425-8ee0-49f1-9604-cbca9324df95.jpeg)

**Comment by croneter** [comment]
What about the others? Issue solved?

**Comment by gregeeh** [comment]
> Is the issue that there are 2 entries in the PMS log instead of 1? Or that the PMS thinks that you are halfway through a video but you're done watching it instead?

The issue is I played the video (TV Show) to the end and did not play anything else and PMS and PKC are telling me I played it once, and then started playing it again for 20 mins.  Which I did not.  Like you have not been able to reproduce it again.

> What about the other users here?! Issue solved?

I thought it was but played a TV Show episode last night, right to the end, and both PMS & PKC are telling me I've not watched it.  Will enable debug now and hopefully it will play up tonight.

**Comment by gregeeh** [comment]
> Is the issue that there are 2 entries in the PMS log instead of 1? Or that the PMS thinks that you are halfway through a video but you're done watching it instead?

The issue is I played one video (TV Show) to the end and both PMS and PKC are telling me a started watching it again straight after I completed watching it.  I supposedly watched it a second time for 20 mins.

 > What about the other users here?! Issue solved?

I thought the issue was solved but watched one TV Show episode last night and both PMS & PKC are telling me it was not watched.  So seems intermittent now.  Will enable debug now and let's hope it plays up.

**Comment by croneter** [comment]
@bjorn1981 @Speederer @gregeeh 
Is this issue solved for you with the brand new PKC version 3.6.5?

**Comment by gregeeh** [comment]
@croneter - Last night I played Hotel Portofino S01E02 at around 7:30PM Feb 2nd.  It has been logged as being played twice, which is not the case.

Log is here:
https://drive.google.com/file/d/1b16gjRmEYHFX09Tw8tRWoMX0AXplx-JJ/view?usp=sharing

Here what Tautulli shows as the play history:
![2022-02-03_08-03-36](https://user-images.githubusercontent.com/46234729/152237110-88c31f43-d108-473b-b7bd-70df8b029f65.jpg)

We did stop and resume the episode but definitely did not play it again after it finished.

**Comment by gregeeh** [comment]
Here is another example of a Movie being played from start to end and being shown in PMS & PKC as being played more then once.  [Log is here.](https://drive.google.com/file/d/18DOEVPL75xXHsbnAiiUrYpdhFDHfT1hR/view?usp=sharing)
![2022-02-04_07-25-20](https://user-images.githubusercontent.com/46234729/152423646-d160bb6a-bba3-48dc-bf29-b10e165858d7.jpg)
![2022-02-04_07-25-52](https://user-images.githubusercontent.com/46234729/152423653-55c3c7fe-efed-4f6a-82ab-4240570dff8d.jpg)

**Comment by bjorn1981** [comment]
I will check tomorrow. The github messages ended up in spam folder, sorry for the late response.

**Comment by croneter** [comment]
@gregeeh 
I still can't replicate your issue, very sorry for that 😞. Also, you seem to be the only one suffering of this. 

Looking at your Tautullli screenshots, each affected video seems to have been played 5+ times. How did your "real" playback look like? Did you pause the video? Did you start and resume the video? 

What skin are you using? How do you start playback exactly? (From Kodi home screen, from movie library, from the detailed movie information page?) How exactly did you resume playback?

**Comment by croneter** [comment]
@gregeeh 
See commenty by koolo on forums.plex.tv:

> I figured this out. That movie or ep is opened and minimized in one of your devices. Which wont allow it to “complete”

 and 

> So. Sometime you would open a movie in plex in an ipad. You would then finish it in pkc. But it will never complete. It will always go back to paused in pkc. The reason is because you stopped it in ipad and never closed it

Is that applicable to your case?

**Comment by bjorn1981** [comment]
Tested with 3.6 7 beta. Playback, stop and resume point is synced. Reset resume point also works. Thank you for the fix.

**Comment by gregeeh** [comment]
@croneter 
Obviously the logs did not reveal anything.  To answer your questions:

How did your "real" playback look like? 
Don't understand this question.  But the playback looked normal.

Did you pause the video? Did you start and resume the video?
As already mentioned, in one example above we did indeed pause and resume.

What skin are you using? 
Estuary

How do you start playback exactly? (From Kodi home screen, from movie library, from the detailed movie information page?)
From the Kodi movie library.

How exactly did you resume playback?
To pause press the pause button on the remote, to resume press the play button on the remote.

Just a thought, I recently installed [Trakt ](https://trakt.tv/dashboard) on the PMS and I wonder if that might be the cause?  I'll uninstall it and see if it happens again.

**Comment by gregeeh** [comment]
@croneter 
I think I have a clue as what might be causing these mysterious items as being watched when I have not watched them at all.  Yesterday I added an email notification to Tautulli to email me when someone stated watching a Movie or TV Show.

Last night I started watching TV, via TvH, and I got an email saying I had started watching a TV Show in PKC.  I then went out for a couple of hours, came home and again watched TV.  Yes, another email about watching the same TV Show in PKC/Plex.

How can this be happening?

**Comment by N8b2KxJJt** [comment]
I'm not sure if this should be its own issue, but playback is also not reported to Plex when using an external player.  I know progress won't work, but versions before 3.5.9 would still report to Plex that the player was active.  Now it doesn't even do that and the episode/movie won't be marked as watched on the Plex side.  After the player closes, I get a message saying "Kodi did not tell us the correct index for video."
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/8008894/kodi.log)

**Comment by xenophobentx** [comment]
> @croneter - Last night I played Hotel Portofino S01E02 at around 7:30PM Feb 2nd. It has been logged as being played twice, which is not the case.
> 
> Log is here: https://drive.google.com/file/d/1b16gjRmEYHFX09Tw8tRWoMX0AXplx-JJ/view?usp=sharing
> 
> Here what Tautulli shows as the play history: ![2022-02-03_08-03-36](https://user-images.githubusercontent.com/46234729/152237110-88c31f43-d108-473b-b7bd-70df8b029f65.jpg)
> 
> We did stop and resume the episode but definitely did not play it again after it finished.

Hello actually i'm seeing duplicates in tautulli aswell but 3 different type of clients are affected: One xbox native app, one LG webos client and have duplicate for plexkodiconnect. It might happen that something went wrong with the newest pms updates. Will monitor and give you updates.

**Comment by xenophobentx** [comment]
Check if you have duplicate tautuli process running!!! Actually i had! that caused the issue:

![image](https://user-images.githubusercontent.com/31575788/152695019-283d13b0-b67b-4b99-84d2-633c45b138fd.png)

**Comment by gregeeh** [comment]
> Check if you have duplicate tautuli process running!!! Actually i had! that caused the issue:
> 
> ![image](https://user-images.githubusercontent.com/31575788/152695019-283d13b0-b67b-4b99-84d2-633c45b138fd.png)

I'm not running Windows but Linux and I don't have more then one Tautulli process.  Just wish @croneter would take a look at this issue.

**Comment by croneter** [comment]
Kindly try the brand new BETA version 3.6.9
